### ISSUE TITLE
[A11Y] Make collapsible fieldset accessible

### ISSUE THREAD

**Issue by herbdool** [backdrop/backdrop-issues#6408]
## Description of the need

Fieldsets that are collapsible should follow the accessibility requirements: https://www.w3.org/WAI/ARIA/apg/patterns/accordion/. We've got `<details>` in core but the Form API still offers collapsible fieldsets.

## Proposed solution

If the fieldset is set to collapsible it should have `aria-expanded` toggled.
